src/js/mqtt.js

```diff
@@ -274,6 +274,11 @@ MqttClient.prototype.disconnect = function(err) {
   this._socket.end();
 };
 
+function qosEqual0(qos) {
+  if (!qos || Number.isNaN(qos)) return false;
+  return qos === 0;
+}
```

Comment:
And I'd like to add a `normalizeOpts()` function to replace this, we should use `(options && options.qos) || 0`.